### PR TITLE
HTTPCLIENT-2047: fixed regression in DefaultHostnameVerifier

### PR DESCRIPTION
**Pull request by ok2c**

### CODE REVIEW

File: httpclient/src/test/java/org/apache/http/conn/ssl/TestDefaultHostnameVerifier.java [hunk 1]
```diff
@@ -417,4 +419,28 @@ public void testExtractCN() throws Exception {
...
+                "host.appspot.com",
+                Collections.singletonList(SubjectName.DNS("*.appspot.com")),
+                publicSuffixMatcher);
+        DefaultHostnameVerifier.matchDNSName(
+                "demo-s3-bucket.s3.eu-central-1.amazonaws.com",
+                Collections.singletonList(SubjectName.DNS("*.s3.eu-central-1.amazonaws.com")),
+                publicSuffixMatcher);
+        DefaultHostnameVerifier.matchDNSName(
+                "hostname-workspace-1.ops.domain.local",
+                Collections.singletonList(SubjectName.DNS("hostname-workspace-1.ops.domain.local")),
```
Comment (michael-o) [inline httpclient/src/test/java/org/apache/http/conn/ssl/TestDefaultHostnameVerifier.java]: Call me stupid, but [this](https://publicsuffix.org/list/public_suffix_list.dat) list does not include the `.local` domain and according to [RFC 6762, section 3](https://tools.ietf.org/html/rfc6762#section-3) this hostname is invalid. only a single DNS label is allowed in the `.local` domain.

Comment (ok2c) [reply httpclient/src/test/java/org/apache/http/conn/ssl/TestDefaultHostnameVerifier.java]: @michael-o I am not sure I understand your objection. The domain name in question might well be invalid but the matching logic should still work, should not it?

Comment (michael-o) [reply httpclient/src/test/java/org/apache/http/conn/ssl/TestDefaultHostnameVerifier.java]: I agree that the matching logic should work, but I don't think we should use something forbidden in our tests. `hostname-workspace-1.ops.local` should do, shouldn't it for the tests?

Comment (ok2c) [reply httpclient/src/test/java/org/apache/http/conn/ssl/TestDefaultHostnameVerifier.java]: @michael-o Sure. That I can do.

**Comment by l3ender** [comment]
I see this is targeted for 4.5.12--is there a release in process for that version yet? If there is anything I can do to help please let me know.

**Comment by ok2c** [comment]
@l3ender I will try to get around to next week.

**Pull request merged**